pkg/middleware/util.go

```diff
@@ -25,6 +26,8 @@ func Gziper() macaron.Handler {
 			return
 		}
 
-		ctx.Invoke(macaronGziper)
+		if _, err := ctx.Invoke(gziper); err != nil {
+			logger.Printf("Invoking gzip handler failed: %s\n", err)
```

Comment:
@marefr Does this look right to you? I'm not sure if this is the right logger to use, but I noticed you could get a logger from the middleware in this way.

Reply:
Not sure. So macaron have some dependency injection support which makes above work if there are a logger registered. But the logger isn't registered until here I think which may be a problem: https://github.com/grafana/grafana/blob/master/pkg/middleware/middleware.go#L76 Or where did you "noticed you could get a logger from the middleware in this way"? 

I think a better solution is to either create a new logger [above](https://github.com/grafana/grafana/pull/19749/files#diff-2dece5362b2c098702e82e6c3507891bR12) or either require `Gziper()` to take a logger as argument

Reply:
@marefr I hoped it might be the case that the middleware had a logger configured, so I wanted to use that if it was the case.

Since I was wrong, I'll instantiate a logger instead.

Reply:
I discovered a pattern in middleware handlers for getting a custom context w/ logger, I applied it also for `Gziper`. Please have a look.

Reply:
Instantiated own Gziper logger instead as you suggested @marefr.